### ISSUE TITLE
Kingdom Builder board game

### ISSUE THREAD

**Issue by Counterfeiter** [deepmind/open_spiel#623]
Hello,

I have implemented the Kingdom Builder board game from Queen Games (there a maybe reserved rights) in python3.

The repo is here: https://github.com/Counterfeiter/KingdomBuilder

I have written this game with the goal to train an alpha zero agent to create a really powerful player, but I failed. =(
I would like to ask smarter guys then me, to build a advanced player. So I could learn why I failed that bad...

**The game:**

- non deterministic (random terrain cards)
- some hidden information (terrain cards of opponents) 
- multiplayer (2 - 5 players)
- hexagon grid board (20x20) (complex CNN)
- board could be build with 2 * 8 * 4 = 48 different combinations
- random rule cards for each new game - changes strategy drastically 
- there is a immediately "gold" reward per move, but following this reward will miss the long term strategy and in fact this "nice feature" is useless

**History of my tries:**
I have forked a multiplayer alpha zero implementation for the agent and made a complete code mess with my tries (sorry for that): https://github.com/Counterfeiter/multiplayer-alphazero/tree/AddKB

I have changed the implementation a lot to fit my needs:
- allow hidden information (states) to be saved and transferred to the stateless game class
- add a modified version of rays rllib MCTS with recursive traversing because the original MCTS implementation made a memory and cpu mess, because of my new game interface
- rework the multiplayer value loss function: The loss is only calculated over players current in the game. My hope was that one agent could play strong with different numbers of opponents. I think mathematically there should be no problem and this would be a nice "investigation"
- rework the overall loss function to weight the value head more then the policy head - because the 4802 logits large policy head has never learned anything
- add chance nodes for the terrain cards
- rework observation variable to bypass the CNN and add some extra information to the CNN following FF directly (save some CNN layers) - add some models for this tests
- rework the CNN hexgon interpretation so that each node in each row has the same neighbors  
- add tensorboard support for some training data parameters

I have trained only on my nvidia laptop GPU (GTX 1660 TI). Could be the missing processing power the bottleneck?

Would be great to get some assessments for more advance people.

I know this topic is not a real issue, but if there is no copy right issue this game would be a really good learning environment. Also this games makes a lot of fun to play :) 

Thanks

Sebastian

**Comment by lanctot** [comment]
@Counterfeiter: apologies that we were not able to addressed this, but I was happy to see more interest in Eurogames!
